crml/transaction-payment/src/constants.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018-2020 Parity Technologies (UK) Ltd. and Centrality Investments Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Transaction Payment Module
+//!
+//! Transaction Payment Customized Error Code Constants
+
+pub(crate) mod error_code {
```

Comment:
Did you try using `decl_error!` for this?

Reply:
The reason why I am not consider using `decl_error!` is `InvalidTransaction::Custom` only accept `u8` and also doughnut has hold some fields https://github.com/plugblockchain/plug-blockchain/blob/d2319b9c98c209be11544e5b05b4797e670c5be6/prml/doughnut/src/constants.rs#L19
so if we use `decl_error!` it will be hard to identify the `code` as for specific module the `error` code start from 1..n

Reply:
Ah yes 👍